### ISSUE TITLE
Version 1.1.1 causes Keepass to crash on Manjaro (Missing Newtonsoft.Json)

### ISSUE THREAD

**Issue by hojjatabdollahi** [Rookiestyle/KeePassOTP#79]
## Overview
I updated to the v1.1.1 version and keepass imported the plugin and crashed. Seems like it is missing Newtonsoft.Json 13.
Here is the output in the terminal:

```
Unhandled Exception:
System.IO.FileNotFoundException: Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies.
File name: 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
  at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context (System.Object state) [0x00007] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00021] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in <62b430b945fa49a19a75382ef03e7bed>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.IO.FileNotFoundException: Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies.
File name: 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
  at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context (System.Object state) [0x00007] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00021] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in <62b430b945fa49a19a75382ef03e7bed>:0 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in <62b430b945fa49a19a75382ef03e7bed>:0
```

## Context
OS: Manjaro
KeePass Version: 2.49
Plugin Version: v1.1.1

**Comment by Rookiestyle** [comment]
Newtonsoft.JSON should be included.

Does v 1.1 work?

**Comment by hojjatabdollahi** [comment]
No. The newest one that works is v1.0.1

**Comment by Rookiestyle** [comment]
That's weird, I just tested it successfully on my VM running Ubuntu and it works just fine.

I'm a bit confused as according to the error message, your system is looking for NewtonSoft 13.0.0.0 although KeePassOTP references 13.0.1
I'll need to figure out why Manjaro does not find Newtonsoft while Ubunto does.

Which version of Manjaro do you use?
Can you by chance check with a different *nix?

**Comment by Rookiestyle** [comment]
1. Downloaded manjaro-xfce-21.1.6-minimal-211017-linux513.iso and added it to my VM
2. Downloaded and installed mono 6.12.0.122-1
2. Downloaded KeePass 2.49 portable
3. Downloaded KeePassOTP 1.1.1
4. Started KeePass with debug parameters `-debugplugin:keepassotp -debugautosave`

No errors during startup
TFA sites are loaded and parsed
No issues observed

Debug file is attached here
If your file looks different, please attach it.
[Debug_KeePassOTP_20211114T195149Z.txt](https://github.com/Rookiestyle/KeePassOTP/files/7534537/Debug_KeePassOTP_20211114T195149Z.txt)

**Comment by hojjatabdollahi** [comment]
Looks like the portable version works. However the one installed from the community repository creates a runner script called `/usr/bin/keepass` that calls `mono` with the `--verify-all` flag. If I remove that flag, I can also run Keepass without any errors. 
But if I add that flag it will crash with the message I posted. 
I added the `--verify-all` flag back and ran `keepass -debugplugin:keepassotp -debugautosave` all the errors that I had in my original post got printed to the terminal, but the Debug looks normal. I've attached it.
 
[Debug_KeePassOTP_20211114T214005Z.txt](https://github.com/Rookiestyle/KeePassOTP/files/7534729/Debug_KeePassOTP_20211114T214005Z.txt)

**Comment by Rookiestyle** [comment]
I'll try later this day with the *verify-all* option and see whether I can find some more details. It seems to be a Newtonsoft thing, maybe a different version behaves different.
If you experience no issues without this option,I propose to not use it for now.

**Comment by Rookiestyle** [comment]
I could reproduce the issue which is related to mono and/or Newtonsoft - nothing I can fix.

Nevertheless, I was able to work around it by using an older version of Newtonsoft, please try the attached version and let me know the result
This version works fine on my Linux VMs - using --verify-all or not makes no difference
[KeePassOTP.zip](https://github.com/Rookiestyle/KeePassOTP/files/7541275/KeePassOTP.zip)

**Comment by hojjatabdollahi** [comment]
Wow. Thanks for working so hard to fix this. 
This version definitely works fine with `--verify-all` flag. I tested it a few times.

I deleted the plugin cache in `~/.local/share/KeePass/PluginCache` and kept adding and removing this plugin by copying it into `/usr/share/keepass/Plugins` and removing it from there. 

Only once, Keepass crashed, I doubt it's related to KeePassOTP but I saw a similar crash yesterday when I was trying to run Keepass with the debug flags, I thought it is because the logger is trying to read/write too much and shrugged it off. But since it happened again without those debug flags, I don't know what could be the cause. 
I attached the log, but it is not important. If you run Keepass again (after the crash) it works fine. So, I tested Keepass and kept running it again, I couldn't recreate this crash. I kept removing the plugin and adding it back, but it didn't crash again (I kept deleting the plugin cache to be sure). 

So, in conclusion, I think your fix works fine and this issue will not happen on the Manjaro package anymore. 
Thank you so much for going through the trouble of setting up a whole VM to figure out the solution.

[log.txt](https://github.com/Rookiestyle/KeePassOTP/files/7541483/log.txt)

**Comment by Rookiestyle** [comment]
Glad it helped.

This adjustment will be included in the next release.
